src/screens/MainApp/MyDiagnoses/index.js

```diff
@@ -159,33 +213,46 @@ const MyDiagnoses = ({ navigation }) => {
 
   return (
     <Background>
-      <HeaderForScrolling show={scrolling} />
-      {diagnoses &&
-        <SwipeListView
-          ref={flatListRef}
-          data={diagnoses}
-          showsVerticalScrollIndicator={false}
-          style={styles.container}
-          ListHeaderComponent={<StaticHeader />}
-          renderItem={({ item }) => <RenderDiagnose animationDuration={DIAGNOSE_ANIMATION_DURATION} item={item} />}
-          keyExtractor={item => item.id}
-          disableVirtualization={false}
-          contentContainerStyle={styles.flatListContainer}
-          onScroll={({ nativeEvent }) => { setScrolling(isScrollWithinSnapThreshold(nativeEvent)) }}
-          renderHiddenItem={(data, rowMap) => (
-            <View style={[styles.hiddenField]}>
-              <SvgButton
-                height={verticalScale(30)}
-                width={scale(21)}
-                svg={trashIcon}
-              />
-            </View>
-          )}
-          disableRightSwipe
-          disableLeftSwipe={!isListSwipeEnabled}
-          onRowOpen={handleListItemSwipe}
-          rightOpenValue={scale(-375)}
-        />}
+      <SwipeListView
+        ref={flatListRef}
+        data={diagnoses}
+        extraData={tab}
+        showsVerticalScrollIndicator={false}
+        style={styles.container}
+        ListHeaderComponent={
+          <View>
+            <StaticHeader />
+            <TabView
+              navigationState={{ index, routes }}
+              renderScene={() => null}
+              renderTabBar={renderTabBar}
+              onIndexChange={setIndex}
+              swipeEnabled={false}
+              lazy
+            />
+            {isEmpty(closedDiagnoses) && !downloadingDiagnoses && tab === 'ResolvedDiagnoses' && <NoDiagnoses tab={tab} />}
+            {isEmpty(openDiagnoses) && !downloadingDiagnoses && tab === 'OpenDiagnoses' && <NoDiagnoses tab={tab} />}
+          </View>
+        }
+        renderItem={({ item }) => <RenderDiagnose animationDuration={DIAGNOSE_ANIMATION_DURATION} item={item} />}
```

Comment:
Please check if you can extract this `renderItem` function outside, otherwise each rerender it's re-created.

For example `renderItem = {renderItem}`

and 

```
const renderItem = useCallback(()=>{
    // return RenderDiagnose
},[])
```